### DISCUSSION TITLE
Environment variables and airflow.cfg mapping

### DISCUSSION THREAD

**Discussion by erosons** [airflow-helm/charts#780] [Questions & Answers]
@mb-m @thesuperzapper referencing readme [https://github.com/airflow-helm/charts/blob/main/charts/airflow/docs/faq/configuration/airflow-configs.md](https://github.com/airflow-helm/charts/pull/url)

if an env is set say this use case AIRFLOW__WEBSERVER__WEB_SERVER_SSL_KEY from a Secret , does the airflow.cfg automatic detect the the env , I see the from docs. it says you can either set in the airflow.cfg or environment variables
image
![image](https://github.com/airflow-helm/charts/assets/73761240/35f06d9a-0f50-4900-b701-6f0ed4532544)

My main concern I don't see theses env in list env in the web server pod, is this env hidden because of security, appreciate contributions.

image

**Comment by thesuperzapper** [comment]
@erosons It's just the way that airflow works. You can set configs with a combination of the `airflow.cfg` file AND/OR the environment variables, but they are not connected in and of themselves.

The chart uses environment variables because they trigger airflow to restart (and pick up the new configs) when the PodSpec's [`envFrom` secret](https://github.com/airflow-helm/charts/blob/main/charts/airflow/templates/config/secret-config-envs.yaml) changes (we put a [hash of the secret contents the Pod's annotations](https://github.com/airflow-helm/charts/blob/c4ade4e5820cd9f3e8d26d6640da04fdff33b9c6/charts/airflow/templates/scheduler/scheduler-deployment.yaml#L49)).

**Reply by erosons** [reply]
so basically this will be right format for to definition in the values.yaml  for the hash to validate correctly , or is there a defacto fromat definition for the env in the values.yaml
```
airflow:
   extraEnvFrom:
     - name: AIRFLOW__WEBSERVER__WEB_SERVER_SSL_KEY
        valueFrom:
           secretKeyRef:
             name: web-cert-and-key
             key: server.key
```
